src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -570,7 +603,7 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 			this._setProcessState(ProcessState.KilledByProcess);
 		}
 
-		this._onProcessExit.fire(exitCode);
+		this._onProcessExit.fire(this.shellIntegrationAttempted ? 1337 : exitCode);
```

Comment:
Should have a constant for this with a comment

Reply:
going with `ShellIntegrationOscPs.Vscode`

Reply:
maybe that's confusing though actually